File: loader/src-tauri/src/shell.rs
use std::process::Command;
use tauri::{
    plugin::{Builder, TauriPlugin},
    Runtime,
};

#[tauri::command]
fn expand_folder(path: &str) {
    #[cfg(windows)]
    Command::new("explorer")
        .args(["/expand,", path])
        .spawn()
        .unwrap();

    #[cfg(target_os = "macos")]
    Command::new("open").arg(path).spawn().unwrap();
}

#[tauri::command]
fn reveal_file(path: &str) {
    #[cfg(windows)]
    Command::new("explorer")
        .args(["/select,", path])
        .spawn()
        .unwrap();

    #[cfg(target_os = "macos")]
    Command::new("open").args(["-R", path]).spawn().unwrap();
}

pub fn init<R: Runtime>() -> TauriPlugin<R> {
    Builder::new("shell")
        .invoke_handler(tauri::generate_handler![expand_folder, reveal_file,])
        .build()
}

File: loader/src/lib/shell.ts
import { invoke, shell } from '@tauri-apps/api'

export const Shell = {
    /**
     * Expand a folder in file explorer.
     * @param path Absolute path to folder.
     */
    async expandFolder(path: string) {
        await invoke('plugin:shell|expand_folder', {
            path: path
        })
    },

    /**
     * Reveal a file in file explorer.
     * @param path Absolute path to file.
     */
    async revealFile(path: string) {
        await invoke('plugin:shell|reveal_file', {
            path: path
        })
    },

    /**
     * Open an external link.
     * @param url URL.
     */
    async openLink(url: string) {
        if (typeof url === 'string' && url.startsWith('https://')) {
            await shell.open(url)
        }
    },
}